diff --git a/src/wsh/Makefile b/src/wsh/Makefile
index 1c5bde2..677c73d 100644
--- a/src/wsh/Makefile
+++ b/src/wsh/Makefile
@@ -7,21 +7,14 @@
 # This file is licensed under MIT License.
 #
 
-OBJLIB := ./lua/src/liblua.a ./openlibm/libopenlibm.a
-
-#CFLAGS := -rdynamic -W -Wall -Wextra -O0 -mpreferred-stack-boundary=12 -mstackrealign -ggdb -g3 -Wno-unused-but-set-variable -Wno-unused-parameter -I./include -rdynamic -I../../include/ -I./luajit-2.0/src/ -Wl,-E -Wl,-z,now
-#OBJLIB := ./luajit-2.0/src/libluajit.a ./openlibm/libopenlibm.a 
-
 all::
-	cd openlibm && make CFLAGS="-fpie -fPIC"
-	cd lua && make linux CFLAGS="-fpie -fPIC"
 	$(CC) $(CFLAGS) wsh.c -o wsh.o -c -fpie -fPIC -ldl -lreadline
 	$(CC) $(CFLAGS) wshmain.c -o wshmain.o -c -fpie -fPIC -ldl -lreadline
 	$(CC) $(CFLAGS) helper.c -o helper.o -c -fpie -fPIC
-	$(CC) $(CFLAGS) linenoise/linenoise.c -o linenoise.o -c -fpie -fPIC
+	$(CC) $(CFLAGS) @linenoise@/src/linenoise.c -o linenoise.o -c -fpie -fPIC
 	$(CC) $(CFLAGS) wsh.o helper.o linenoise.o -shared -fPIC -o libwitch.so
-	ar cr libwitch.a wsh.o helper.o linenoise.o
-	$(CC) $(CFLAGS) wsh.o helper.o linenoise.o wshmain.o -o wsh -Wl,-T -Wl,script.lds -liberty $(OBJLIB) -ldl
+	$(AR) cr libwitch.a wsh.o helper.o linenoise.o
+	$(CC) $(CFLAGS) wsh.o helper.o linenoise.o wshmain.o -o wsh -Wl,-T -Wl,script.lds -liberty @luajit@/lib/libluajit-5.1.a @openlibm@/lib/libopenlibm.a -ldl
 
 	cp wsh ../../bin/
 
@@ -30,13 +23,9 @@ test:
 
 clean::
 	rm wsh helper.o wsh.o wshmain.o libwitch.so libwitch.a linenoise.o learnwitch.log -f
-	cd openlibm && make clean
-	cd lua && make clean
 	cd tests && make clean
 
 deepclean:
-	cd openlibm && make clean
-	cd lua && make clean
 	make clean
 
 install::
diff --git a/src/wsh/script.lds b/src/wsh/script.lds
index 05a3fc6..d5ea980 100644
--- a/src/wsh/script.lds
+++ b/src/wsh/script.lds
@@ -7,7 +7,7 @@ OUTPUT_FORMAT("elf64-x86-64", "elf64-x86-64",
 	      "elf64-x86-64")
 OUTPUT_ARCH(i386:x86-64)
 ENTRY(_start)
-SEARCH_DIR("=/usr/x86_64-linux-gnu/lib64"); SEARCH_DIR("=/usr/local/lib64"); SEARCH_DIR("=/lib64"); SEARCH_DIR("=/usr/lib64"); SEARCH_DIR("=/usr/x86_64-linux-gnu/lib"); SEARCH_DIR("=/usr/local/lib"); SEARCH_DIR("=/lib"); SEARCH_DIR("=/usr/lib");
+SEARCH_DIR("=@luajit@/lib");
 SECTIONS
 {
   /* Read-only sections, merged into text segment: */
